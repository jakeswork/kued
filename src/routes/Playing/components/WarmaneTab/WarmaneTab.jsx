import React from 'react';
import { Tabs, TabList, TabPanel } from 'react-tabs';

import { propTypes, defaultProps } from './types';
import ServerTabLayout from '../../../../layouts/ServerTabLayout';
import TabHeading from '../TabHeading';

const options = [
  {
    value: '2v2',
    label: '2v2',
  },
  {
    value: '3v3',
    label: '3v3',
  },
];

const WarmaneTab = ({ classes, ...props }) => (
  <TabPanel {...props}>
    <Tabs selectedTabClassName={classes.tabActive}>
      <TabList className={classes.tabList}>
        <TabHeading sub>TBC</TabHeading>
        <TabHeading sub>WoTLK</TabHeading>
      </TabList>
      <ServerTabLayout
        dropdownOptions={options}
        dropdownOptionDefault={options[0]}
      >
        TBC
      </ServerTabLayout>
      <ServerTabLayout
        dropdownOptions={options}
        dropdownOptionDefault={options[0]}
      >
        WoTLK
      </ServerTabLayout>
    </Tabs>
  </TabPanel>
);

WarmaneTab.propTypes = propTypes;

WarmaneTab.defaultProps = defaultProps;

WarmaneTab.tabsRole = 'TabPanel';

export default WarmaneTab;
