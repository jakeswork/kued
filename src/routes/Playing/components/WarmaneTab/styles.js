export default theme => ({
  tabActive: {
    color: theme.colorPrimary,
    cursor: 'default',
    pointerEvents: 'none',
  },
  tabList: {
    display: 'flex',
  },
});
