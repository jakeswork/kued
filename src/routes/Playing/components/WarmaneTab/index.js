import withStyles from 'react-jss';

import WarmaneTab from './WarmaneTab';
import styles from './styles';

export default withStyles(styles)(WarmaneTab);
