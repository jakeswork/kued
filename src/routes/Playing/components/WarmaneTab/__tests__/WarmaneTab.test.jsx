import React from 'react';
import { shallow } from 'enzyme';

import WarmaneTab from '../WarmaneTab';

const defaultProps = {
  classes: {},
};

describe('The WarmaneTab instance', () => {
  describe('When called with default props', () => {
    it('should render without throwing an error', () => {
      const wrapper = shallow(<WarmaneTab {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
