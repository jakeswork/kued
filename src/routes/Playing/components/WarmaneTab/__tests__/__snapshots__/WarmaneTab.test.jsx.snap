// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The WarmaneTab instance When called with default props should render without throwing an error 1`] = `
<TabPanel
  className="react-tabs__tab-panel"
  forceRender={false}
  selectedClassName="react-tabs__tab-panel--selected"
>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <TabList
      className="react-tabs__tab-list"
    >
      <Jss(TabHeading)
        sub={true}
      >
        TBC
      </Jss(TabHeading)>
      <Jss(TabHeading)
        sub={true}
      >
        WoTLK
      </Jss(TabHeading)>
    </TabList>
    <Jss(ServerTabLayout)
      dropdownOnChange={[Function]}
      dropdownOptionDefault={
        Object {
          "label": "2v2",
          "value": "2v2",
        }
      }
      dropdownOptions={
        Array [
          Object {
            "label": "2v2",
            "value": "2v2",
          },
          Object {
            "label": "3v3",
            "value": "3v3",
          },
        ]
      }
    >
      TBC
    </Jss(ServerTabLayout)>
    <Jss(ServerTabLayout)
      dropdownOnChange={[Function]}
      dropdownOptionDefault={
        Object {
          "label": "2v2",
          "value": "2v2",
        }
      }
      dropdownOptions={
        Array [
          Object {
            "label": "2v2",
            "value": "2v2",
          },
          Object {
            "label": "3v3",
            "value": "3v3",
          },
        ]
      }
    >
      WoTLK
    </Jss(ServerTabLayout)>
  </Tabs>
</TabPanel>
`;
